### PR TITLE
feat: The `json` module is now avaiable to stream maps

### PR DESCRIPTION
**Pull request by grigi**
This allows using `json.loads()` and `json.dumps()` to convert fields between text and json representations.

Particularily useful moving data between MySQL and Postgres where MySQL stores json as text, and Postgres stores json as JSONB.

Usage in a stream_map:
```yaml
        stream_maps:
          sample-reports:
            recipients_email_addresses: json.loads(recipients_email_addresses)
```

Closes https://github.com/meltano/sdk/issues/1502

----
📚 Documentation preview 📚: https://meltano-sdk--2531.org.readthedocs.build/en/2531/

### CODE REVIEW

**Comment by grigi** [comment]
Ah, you want me to add docs to https://meltano-sdk--2531.org.readthedocs.build/en/2531/stream_maps.html#built-in-functions ?

**Comment by edgarrmondragon** [comment]
> Ah, you want me to add docs to https://meltano-sdk--2531.org.readthedocs.build/en/2531/stream_maps.html#built-in-functions ?

yeah 🙏

File: docs/stream_maps.md [hunk 1]
```diff
@@ -234,6 +234,8 @@ can be referenced directly by mapping expressions.
     `hashlib.md5(<input>.encode("utf-8")).hexdigest()`.
 - `datetime` - This is the datetime module object from the Python standard library. You can access
     datetime.datetime, datetime.timedelta, etc.
+- `json` - This is the json module object from the Python standard library.
+    Primarily used for calling json.dumps() and json.loads().
```
Comment (edgarrmondragon) [inline docs/stream_maps.md:238]:
```suggestion
- `json` - This is the json module object from the Python standard library.
    Primarily used for calling :py:func:`json.dumps()` and :py:func:`json.loads()`.
```

File: docs/stream_maps.md [hunk 2]
```diff
@@ -234,6 +234,8 @@ can be referenced directly by mapping expressions.
     `hashlib.md5(<input>.encode("utf-8")).hexdigest()`.
 - `datetime` - This is the datetime module object from the Python standard library. You can access
     datetime.datetime, datetime.timedelta, etc.
+- `json` - This is the json module object from the Python standard library.
+    Primarily used for calling :py:func:`json.dumps()` and :py:func:`json.loads()`.
```
Comment (edgarrmondragon) [inline docs/stream_maps.md:238]:
```suggestion
    Primarily used for calling [`json.dumps()`](inv:py:func:json.dumps) and [`json.loads()`](inv:py:func:json.loads).
```

File: docs/stream_maps.md [hunk 3]
```diff
@@ -234,6 +234,8 @@ can be referenced directly by mapping expressions.
     `hashlib.md5(<input>.encode("utf-8")).hexdigest()`.
 - `datetime` - This is the datetime module object from the Python standard library. You can access
     datetime.datetime, datetime.timedelta, etc.
+- `json` - This is the json module object from the Python standard library.
+    Primarily used for calling [`json.dumps()`](inv:py:func:json.dumps) and [`json.loads()`](inv:py:func:json.loads).
```
Comment (edgarrmondragon) [inline docs/stream_maps.md:238]:
```suggestion
    Primarily used for calling [`json.dumps()`](inv:python:py:function:#json.dumps)
    and [`json.loads()`](inv:python:py:function:#json.loads).
```

**Review by edgarrmondragon — APPROVED** [review]

**Comment by grigi** [comment]
Thanks for those documentation improvements @edgarrmondragon :pray:

File: docs/stream_maps.md [hunk 4]
```diff
@@ -234,6 +234,9 @@ can be referenced directly by mapping expressions.
     `hashlib.md5(<input>.encode("utf-8")).hexdigest()`.
 - `datetime` - This is the datetime module object from the Python standard library. You can access
     datetime.datetime, datetime.timedelta, etc.
+- `json` - This is the json module object from the Python standard library.
+    Primarily used for calling [`json.dumps()`](inv:python:py:function:#json.dumps)
```
Comment (edgarrmondragon) [inline docs/stream_maps.md:238]:
```suggestion
- [`json`](inv:python:py:module:#json) - This is the json module object from the Python standard
    library. Primarily used for calling [`json.dumps()`](inv:python:py:function:#json.dumps)
```

**Comment by edgarrmondragon** [comment]
Thanks @grigi!

**Pull request merged**